src/test/suite/completion.test.ts

```diff
@@ -438,6 +438,31 @@ describe("Should do completion", () => {
       stdinMock.write(new SuggestionShownRequestMatcher("dom"), "utf8")
     ).once();
   });
+  it("should report suggestion shown only once for the same suggestion", async () => {
+    mockAutocomplete(
```

Comment:
I'm not sure, but isn't the response is:
```
      anAutocompleteResponse("blibliblid", "ata"),
      anAutocompleteResponse("blibliblida", "ta"),
      anAutocompleteResponse("blibliblidat", "a")
```
?

Reply:
@yairco1990 you're right and wrong - as long as its the same word, the `new_prefix` will remain the same, and the `old_suffix` will change.
But if you type beyond the first word, the `new_prefix` will change as well.
@dimacodota wdyt? this means that you'll still fire the shown even twice for the same logical completion

Reply:
if we are not rendering a new completion in this case it is an issue

Reply:
not exactly,  the binary will return the common text between the current text and the suggestion as `old_prefix` that should be replaced: see an example: 
```
"const da"  -> {"old_prefix":"da","results":[{"new_prefix":"data =" ...}]
"const dat"  -> {"old_prefix":"dat","results":[{"new_prefix":"data ="...}]
```

but there are places(especially after whitespaces ` `) that although it is the same suggestion, it will not contain the  `old_prefix`  like in the following example:
```
"const data" -> {"old_prefix":"data","results":[{"new_prefix":"data = new"..}]
"const data " -> {"old_prefix":"","results":[{"new_prefix":"= new"..}]
```
added a fix for the second case(ie the prev suggestion end with the current)

Reply:
handled here https://github.com/codota/tabnine-vscode/pull/1003/commits/1fdeb81cdb4a6e1660b5430906ecac327be350dc

Reply:
great 👍
and what happens if the suffix has changed? doesn't it mean that we just rendered a new completion but we don't fire the event?

Reply:
Of course im missing the "not" there

Reply:
@yairco1990 - can you please elaborate on what you've meant?   the event should be fired in case the prev suggestion is not ending with the current suggestion

Reply:
what I meant is that you recognize there is no change in the suggestion based only on the prefix. my question is if there is an option this condition is evaluated to "true", but the suffix has changed, and therefore, it should be counted as a new suggestion.

Reply:
oh you mean if the `new_suffix` will be different?

Reply:
yep

Reply:
oh, the `new_suffix` handles the closing brackets in single-line suggestions,  not sure how common is the case of the same suggestion w/o closing brackets. so ill merge it as is